queries/vrl/highlights.scm

```diff
@@ -0,0 +1,118 @@
+(comment) @comment
+
+[
+  (null)
+  (boolean)
+] @constant.builtin
+
+(timestamp) @constant
+
+(closure_variables
+  "|" @punctuation.bracket)
+
+(closure_variables
+  (ident) @variable.parameter)
+
+(integer) @number
+
+(float) @number
+
+[
+  (string)
+  (raw_string)
+] @string
+
+[
+  (raw_string_escape_sequence)
+  (escape_sequence)
+  (regex_escape_sequence)
+] @string.escape
+
+(string_template
+  (ident) @variable)
+
+(string_template
+  "{{" @punctuation.special
+  (_)
+  "}}" @punctuation.special)
+
+(regex) @string.regexp
+
+(boolean) @boolean
+
+(ident) @variable
+
+(event) @variable.builtin
+
+(function_call
+  (ident) @function.call)
+
+; VRL queries
+(query
+  (path
+    [
+      (field) @variable
+      (string) @string
+      (index) @number
+    ]))
+
+[
+  "abort"
+  ; "as"
+  ; "break"
+  ; "continue"
+  "else"
+  "false"
+  ; "for"
+  "if"
+  ; "impl"
+  ; "in"
+  ; "let"
+  ; "loop"
+  "null"
+  "return"
+  ; "self"
+  ; "std"
+  ; "then"
+  ; "this"
+  "true"
+  ; "type"
+  ; "until"
+  ; "use"
+  ; "while"
+] @keyword
+
+[
+  "="
+  "=="
+  "!="
+  "|="
+  ">"
+  ">="
+  "<"
+  "<="
+  "+"
+  "-"
+  "*"
+  "/"
+  "&&"
+  "||"
+  "??"
+  "|"
+  "!"
+  "."
```

Comment:
Thanks, @ribru17.

Btw, how can I deal with the `"."` used as both query path delimiter and the default builtin `event` variable?

Reply:
I think I solved this by adding predicates to the `event` query. But that's a little weird.

Reply:
Just do
```
((path)
  "." @punctuation.delimiter)
```

to only capture the "." within `(path)`.

Reply:
That's a simpler query, but the `"."` in `(query)` will match both of the two queries. It only works for highlighting, because seems the @punctuation.delimiter priority is higher than @variable.builtin.

For variable reference, the `"."` in `(query)` will be select as `event` references if  the `(event)` query is not predicated with `#not-has-parent?`.

<img width="550" alt="image" src="https://github.com/user-attachments/assets/60559fd0-55a6-4d87-86c1-fbf2c2903d26">

I don't want the interior points to be visible in the tree. Do I have to, to prevent `not-has-parent` in `locals.scm`?

Reply:
Not in my tests; there the first dot and the following dots were separately captured as (I believe) expected.

Reply:
<img width="776" alt="Screenshot 2024-07-30 at 19 03 54" src="https://github.com/user-attachments/assets/2c17fd15-cd57-4ee9-b909-400c79a12280">
<img width="776" alt="Screenshot 2024-07-30 at 19 04 03" src="https://github.com/user-attachments/assets/cb0092c6-55cf-459e-a0fe-81b3e9fc56fa">

Reply:
(I'm less concerned about locals, as these are currently not really used in nvim-treesitter.)

Reply:
Resolved by putting `(event)` inside `(query)`. Thanks, @clason :)

The variable reference highlights feature is provided by [vim-illuminate](https://github.com/RRethy/vim-illuminate/blob/master/lua/illuminate/providers/treesitter.lua#L16), which is useful if there is not an LSP available.